lib/go-parser/symbols/index.ts

```diff
@@ -0,0 +1,203 @@
+        const fileoff = this.binary.Uint32(functab.slice(functabsize));
+        this.filetab = b.slice(Number(fileoff));
+        this.nfiletab = Number(this.binary.Uint32(this.filetab));
+        break;
+      default:
+        throw new Error("unreachable");
+    }
+  }
+
+  // go12MapFiles returns a list of files that have been found in the symbol table.
+  // In the original Go implementation, this function takes a map and object, but
+  // we don't need to construct a map and don't need the object.
+  // https://cs.opensource.google/go/go/+/refs/tags/go1.18.5:src/debug/gosym/pclntab.go;l=669
+  public go12MapFiles(): string[] {
+    this.initFileMap();
+    const m: string[] = [];
+    for (const file of Object.keys(this.fileMap)) {
+      m.push(file);
+    }
+    return m;
+  }
+
+  // uintptr returns the pointer-sized value encoded at b.
+  // The pointer size is dictated by the table being read.
+  private uintptr(b: Buffer): bigint {
+    if (this.ptrsize === 4) {
+      return BigInt(this.binary.Uint32(b));
+    }
+    return this.binary.Uint64(b);
+  }
+
+  // functabFieldSize returns the sivze in bytes of a single functab field.
+  // https://cs.opensource.google/go/go/+/refs/tags/go1.18.5:src/debug/gosym/pclntab.go;l=377
+  private functabFieldSize(): number {
+    if (this.version >= pclnVersion.v118) {
+      return 4;
+    }
+    return this.ptrsize;
+  }
+
+  // string returns a Go string found at offset.
+  // https://cs.opensource.google/go/go/+/refs/tags/go1.18.5:src/debug/gosym/pclntab.go;l=372
+  private string(offset: number): string {
+    return this.stringFrom(this.funcdata, offset);
+  }
+
+  // initFileMap initializes the map from file name to file number.
+  // We technically don't need the number, but match Go's implementation 1:1 instead.
+  // https://cs.opensource.google/go/go/+/refs/tags/go1.18.5:src/debug/gosym/pclntab.go;l=641
+  private initFileMap() {
+    if (this.fileMap.size > 0) {
+      return;
+    }
+
+    const m: Map<string, number> = new Map();
+    if (this.version === pclnVersion.v12) {
+      for (let i = 1; i < this.nfiletab; i++) {
+        const s = this.string(
+          Number(this.binary.Uint32(this.filetab.slice(4 * i))),
+        );
+        m[s] = i;
+      }
+    } else {
+      let pos: number = 0;
+      for (let i = 0; i < this.nfiletab; i++) {
+        const s = this.stringFrom(this.filetab, pos);
+        m[s] = pos;
+        pos += s.length + 1;
+      }
+    }
+    this.fileMap = m;
+  }
+
+  // stringFrom returns a Go string found at offset from a position.
+  // https://cs.opensource.google/go/go/+/refs/tags/go1.18.5:src/debug/gosym/pclntab.go;l=361
+  private stringFrom(arr: Buffer, offset: number): string {
+    const i = arr.slice(offset).indexOf(0);
+    const s = arr.slice(offset, offset + i).toString("ascii");
+    return s;
+  }
+}
+
+enum pclnVersion {
+  unknown,
+  v11,
+  v12,
+  v116,
+  v118,
```

Comment:
will future Go versions require us to modify this code?

Reply:
Potentially, yes. I wanted to look into the cronjob you added that builds binaries with the latest version & checks compatibility, and integrate this parser into these tests as well. In a separate PR, probably :)